NitroxPatcher/Patches/Dynamic/PrisonPredatorSwimToPlayer_Evaluate_Patch.cs

```diff
@@ -0,0 +1,116 @@
+using System.Collections.Generic;
+using System.Reflection;
+using System.Reflection.Emit;
+using HarmonyLib;
+using NitroxClient.GameLogic;
+using NitroxClient.GameLogic.PlayerLogic;
+using NitroxModel.DataStructures;
+using NitroxModel.Helper;
+using UnityEngine;
+
+namespace NitroxPatcher.Patches.Dynamic;
+
+public sealed partial class PrisonPredatorSwimToPlayer_Evaluate_Patch : NitroxPatch, IDynamicPatch
+{
+    internal static readonly MethodInfo TARGET_METHOD = Reflect.Method((PrisonPredatorSwimToPlayer t) => t.Evaluate(default(Creature), default(float)));
+
+    internal static readonly EcoRegion.TargetFilter isTargetValidFilter = new(IsTargetValid);
+
+    /// <summary>
+    /// Replace all the method with our custom <see cref="EvaluatePriority(PrisonPredatorSwimToPlayer, Creature, float)" />
+    /// 
+    /// Original method does hardcode Player.main usage and doesn't use any "GameObject" or Target abstraction as in other <see cref="CreatureAction"/>
+    /// So we need to rewrite <see cref="PrisonPredatorSwimToPlayer.Evaluate(Creature, float)"/> to use the same logic as <see cref="PrisonPredatorSwimToPlayer.Perform(Creature, float)"/>"/>
+    /// </summary>
+    public static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions)
+    {
+        yield return new CodeInstruction(OpCodes.Ldarg_0);
+        yield return new CodeInstruction(OpCodes.Ldarg_1);
+        yield return new CodeInstruction(OpCodes.Ldarg_2);
+        yield return new CodeInstruction(OpCodes.Call, Reflect.Method(() => EvaluatePriority(default, default, default)));
+        yield return new CodeInstruction(OpCodes.Ret);
```

Comment:
Like in `PrisonPredatorSwimToPlayer_Perform_Patch` I think we can manipulated the IL to get the same outcome.
Creating a new local variable, filling it with the ecoTarget and then replacing all usings of the Player local variable with the new one should work.